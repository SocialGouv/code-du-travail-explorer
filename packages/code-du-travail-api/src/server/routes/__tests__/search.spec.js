const request = require("supertest");
const Koa = require("koa");
const routes = require("../search");

const app = new Koa();
app.use(routes.routes());

test("return search results for demission", async () => {
  const response = await request(app.callback()).get(
    `/api/v1/search?q=démission`
  );
  expect(response.status).toBe(200);
  expect(response.body.hits).toMatchSnapshot();
});

test("return faq search results for demission ", async () => {
  const excludeSources = [
    "code_du_travail",
    "fiche",
    "modele_courrier",
    "idcc",
    "kali",
    "themes",
    "fiches_ministere_travail",
    "fiches_service_public"
  ];
  const response = await request(app.callback()).get(
    `/api/v1/search?q=démission&excludeSources=${excludeSources.join(",")}`
  );
  expect(response.status).toBe(200);
  expect(response.body.hits).toMatchSnapshot();
});
