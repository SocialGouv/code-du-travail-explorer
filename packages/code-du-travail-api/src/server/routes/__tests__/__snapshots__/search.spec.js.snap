// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`return article results when searching with article id 1`] = `
Object {
  "facets": Array [
    Object {
      "doc_count": 1,
      "key": "code_du_travail",
    },
  ],
  "hits": Object {
    "hits": Array [
      Object {
        "_id": "8",
        "_index": "cdtn_document_test",
        "_score": 7.2165623,
        "_source": Object {
          "slug": "article-r1225-18",
          "source": "code_du_travail",
          "title": "Article R1225-18",
          "url": "https://www.legifrance.gouv.fr/affichCodeArticle.do?idArticle=LEGIARTI000018537778&cidTexte=LEGITEXT000006072050",
        },
        "_type": "cdtn_document_test",
        "highlight": Object {
          "title.article_id": Array [
            "Article <mark>R1225-18</mark>",
          ],
          "title.french": Array [
            "Article <mark>R1225</mark>-<mark>18</mark>",
          ],
        },
      },
    ],
    "max_score": 7.2165623,
    "total": 1,
  },
}
`;

exports[`return faq search results for demission  1`] = `
Object {
  "hits": Array [
    Object {
      "_id": "6",
      "_index": "cdtn_document_test",
      "_score": 2.3778663,
      "_source": Object {
        "slug": "dans-les-hcr-quel-est-le-preavis-de-demission",
        "source": "faq",
        "title": "Dans les HCR, quel est le préavis de démission ?",
      },
      "_type": "cdtn_document_test",
      "highlight": Object {
        "text.french": Array [
          "Le préavis de <mark>démission</mark> diffère suivant la",
        ],
        "title.french": Array [
          "Dans les HCR, quel est le préavis de <mark>démission</mark>",
        ],
      },
    },
    Object {
      "_id": "7",
      "_index": "cdtn_document_test",
      "_score": 2.1108572,
      "_source": Object {
        "slug": "joccupe-un-poste-de-vendeuse-en-boulangerie-patisserie-et-je-viens-de-poser-ma-demission-quelles-sont-les-consequences-dun-arret-maladie-pendant-la-periode-de-preavis-est-ce-que-je-peux-etre-sanctionnee-en-cas-darret-de-travail",
        "source": "faq",
        "title": "J'occupe un poste de vendeuse en boulangerie-pâtisserie et je viens de poser ma démission. Quelles sont les conséquences d’un arrêt maladie pendant la période de préavis ? Est-ce que je peux être sanctionnée en cas d’arrêt de travail ?",
      },
      "_type": "cdtn_document_test",
      "highlight": Object {
        "text.french": Array [
          "en arrêt maladie pendant le préavis de <mark>démission</mark>",
        ],
        "title.french": Array [
          "boulangerie-pâtisserie et je viens de poser ma <mark>démission</mark>",
        ],
      },
    },
  ],
  "max_score": 2.3778663,
  "total": 2,
}
`;

exports[`return faq search results for smic with snippet content 1`] = `
Object {
  "facets": Array [
    Object {
      "doc_count": 1,
      "key": "fiches_ministere_travail",
    },
  ],
  "hits": Object {
    "hits": Array [
      Object {
        "_id": "15",
        "_index": "cdtn_document_test",
        "_score": 16.334557,
        "_source": Object {
          "anchor": "#A-qui-s-applique-le-SMIC",
          "slug": "le-smic-a-qui-sapplique-le-smic",
          "source": "fiches_ministere_travail",
          "title": "Le SMIC : à qui s’applique le SMIC ?",
          "url": "https://travail-emploi.gouv.fr/droit-du-travail/la-remuneration/article/le-smic-374531#A-qui-s-applique-le-SMIC",
        },
        "_type": "cdtn_document_test",
        "highlight": Object {
          "text.french": Array [
            "Doit percevoir un salaire au moins égal au <mark>SMIC</mark>",
            "conditions de droit privé.Un taux réduit du <mark>SMIC</mark>",
            "n’est pas contrôlable (certains VRP).Le <mark>SMIC</mark>",
            "emplois d’avenir.Sont exclus du bénéfice du <mark>SMIC</mark>",
          ],
          "title.french": Array [
            "Le <mark>SMIC</mark> : à qui s’applique le <mark>SMIC</mark> ?",
          ],
        },
      },
    ],
    "max_score": 16.334557,
    "total": 2,
  },
  "snippet": Object {
    "_id": "14",
    "_index": "cdtn_document_test",
    "_score": 14.235261,
    "_source": Object {
      "html": "<p>Au 1er janvier 2019, le montant du <accronym title='Salaire minimum interprofessionnel de croissance' lang='fr'>SMIC</accronym> brut horaire s'établit à <b>10,03&nbsp;€</b> en métropole, en Guadeloupe, en Guyane, en Martinique, à La Réunion, à Saint-Barthélemy, à Saint-Martin et à Saint-Pierre-et-Miquelon.<br/> Soit <b>1521,22&nbsp;€</b> mensuels sur la base de la durée légale du travail de 35 heures hebdomadaires</p>",
      "references": Array [
        Object {
          "titre": "Décret n° 2018-1173 du 19 décembre 2018 portant relèvement du salaire minimum de croissance ",
          "url": "https://www.legifrance.gouv.fr/eli/decret/2018/12/19/MTRX1833925D/jo/texte",
        },
      ],
    },
    "_type": "cdtn_document_test",
  },
}
`;

exports[`return search results for demission 1`] = `
Object {
  "hits": Array [
    Object {
      "_id": "2",
      "_index": "cdtn_document_test",
      "_score": 3.4251862,
      "_source": Object {
        "anchor": "#Comment-presenter-une-demission",
        "slug": "la-demission-comment-presenter-une-demission",
        "source": "fiches_ministere_travail",
        "title": "La démission : comment présenter une démission ?",
        "url": "https://travail-emploi.gouv.fr/droit-du-travail/la-rupture-du-contrat-de-travail/article/la-demission#Comment-presenter-une-demission",
      },
      "_type": "cdtn_document_test",
      "highlight": Object {
        "text.french": Array [
          "La <mark>démission</mark> ne peut être exigée par avance",
          "aucune forme particulière pour présenter sa <mark>démission</mark>",
          "collectives prévoient d’ailleurs que la <mark>démission</mark>",
          "travail, le salarié a intérêt à remettre sa <mark>démission</mark>",
          "les contestations sur l’intention même de <mark>démissionner</mark>",
        ],
        "title.french": Array [
          "La <mark>démission</mark> : comment présenter une <mark>démission</mark>",
        ],
      },
    },
    Object {
      "_id": "4",
      "_index": "cdtn_document_test",
      "_score": 3.2924087,
      "_source": Object {
        "slug": "demission-dun-salarie",
        "source": "fiches_service_public",
        "title": "Démission d'un salarié",
        "url": "https://www.service-public.fr/particuliers/vosdroits/F2883",
      },
      "_type": "cdtn_document_test",
      "highlight": Object {
        "text.french": Array [
          "Volonté de <mark>démissionner</mark> Pour qu'une <mark>démission</mark>",
          "Volonté de <mark>démissionner</mark> Pour qu'une <mark>démission</mark>",
          "La <mark>démission</mark> vous permet de rompre votre",
          "Pour qu'une <mark>démission</mark> soit valable, vous",
          "Lettre de <mark>démission</mark> du salarié Direction",
        ],
        "title.french": Array [
          "<mark>Démission</mark> d'un salarié",
        ],
      },
    },
    Object {
      "_id": "0",
      "_index": "cdtn_document_test",
      "_score": 3.103135,
      "_source": Object {
        "slug": "themes/rupture-de-contrat/la-demission",
        "source": "themes",
        "title": "La démission",
      },
      "_type": "cdtn_document_test",
      "highlight": Object {
        "text.french": Array [
          "Vous souhaitez <mark>démissionner</mark>?",
        ],
        "title.french": Array [
          "La <mark>démission</mark>",
        ],
      },
    },
    Object {
      "_id": "5",
      "_index": "cdtn_document_test",
      "_score": 2.9205968,
      "_source": Object {
        "slug": "demission-dune-assistante-maternelle",
        "source": "fiches_service_public",
        "title": "Démission d'une assistante maternelle",
        "url": "https://www.service-public.fr/particuliers/vosdroits/F33164",
      },
      "_type": "cdtn_document_test",
      "highlight": Object {
        "text.french": Array [
          "Lettre de <mark>démission</mark> du salarié Direction",
          "Envoi d'une lettre de <mark>démission</mark> L'assistante",
          "son contrat de travail en présentant sa <mark>démission</mark>",
          "présentation de la lettre recommandée de <mark>démission</mark>",
        ],
        "title.french": Array [
          "<mark>Démission</mark> d'une assistante maternelle",
        ],
      },
    },
    Object {
      "_id": "6",
      "_index": "cdtn_document_test",
      "_score": 2.3778663,
      "_source": Object {
        "slug": "dans-les-hcr-quel-est-le-preavis-de-demission",
        "source": "faq",
        "title": "Dans les HCR, quel est le préavis de démission ?",
      },
      "_type": "cdtn_document_test",
      "highlight": Object {
        "text.french": Array [
          "Le préavis de <mark>démission</mark> diffère suivant la",
        ],
        "title.french": Array [
          "Dans les HCR, quel est le préavis de <mark>démission</mark>",
        ],
      },
    },
    Object {
      "_id": "8",
      "_index": "cdtn_document_test",
      "_score": 2.3033223,
      "_source": Object {
        "slug": "article-r1225-18",
        "source": "code_du_travail",
        "title": "Article R1225-18",
        "url": "https://www.legifrance.gouv.fr/affichCodeArticle.do?idArticle=LEGIARTI000018537778&cidTexte=LEGITEXT000006072050",
      },
      "_type": "cdtn_document_test",
      "highlight": Object {
        "path": Array [
          "Paternité/Congés d'éducation des enfants/<mark>Démission</mark>",
        ],
        "text.french": Array [
          "Le salarié informe son employeur de sa <mark>démission</mark>",
        ],
      },
    },
    Object {
      "_id": "7",
      "_index": "cdtn_document_test",
      "_score": 2.1108572,
      "_source": Object {
        "slug": "joccupe-un-poste-de-vendeuse-en-boulangerie-patisserie-et-je-viens-de-poser-ma-demission-quelles-sont-les-consequences-dun-arret-maladie-pendant-la-periode-de-preavis-est-ce-que-je-peux-etre-sanctionnee-en-cas-darret-de-travail",
        "source": "faq",
        "title": "J'occupe un poste de vendeuse en boulangerie-pâtisserie et je viens de poser ma démission. Quelles sont les conséquences d’un arrêt maladie pendant la période de préavis ? Est-ce que je peux être sanctionnée en cas d’arrêt de travail ?",
      },
      "_type": "cdtn_document_test",
      "highlight": Object {
        "text.french": Array [
          "en arrêt maladie pendant le préavis de <mark>démission</mark>",
        ],
        "title.french": Array [
          "boulangerie-pâtisserie et je viens de poser ma <mark>démission</mark>",
        ],
      },
    },
    Object {
      "_id": "9",
      "_index": "cdtn_document_test",
      "_score": 2.0272236,
      "_source": Object {
        "slug": "article-r1442-22-4",
        "source": "code_du_travail",
        "title": "Article R1442-22-4",
        "url": "https://www.legifrance.gouv.fr/affichCodeArticle.do?idArticle=LEGIARTI000033748111&cidTexte=LEGITEXT000006072050",
      },
      "_type": "cdtn_document_test",
      "highlight": Object {
        "text.french": Array [
          "désire renoncer à son mandat adresse sa  <mark>démission</mark>",
          "La <mark>démission</mark>  n'est définitive qu'après acceptation",
        ],
      },
    },
  ],
  "max_score": 3.4251862,
  "total": 8,
}
`;
